### ISSUE TITLE
tsscplatform-jenkins  failure shows in the Overview from the operator.

### ISSUE THREAD

**Issue by piggyvenus** [ploigos/ploigos-software-factory-operator#140]
Installed Ploigos Operator 0.22 on OCP 4.6.15
Got error from the TsscPlatform> Overview > Conditions
The task includes an option with an undefined variable. The error was: list object has no element 0

The error appears to be in '/opt/ansible/roles/tssc-platform/deployment/tasks/main.yml': line 99, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

- set_fact:

Logs from controller manager:
TASK [set_fact] ******************************** 
[0;31mfatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: list object has no element 0\n\nThe error appears to be in '/opt/ansible/roles/tssc-platform/deployment/tasks/main.yml': line 99, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- set_fact:\n  ^ here\n"}[0m

-------------------------------------------------------------------------------
{"level":"error","ts":1613766230.2726583,"logger":"logging_event_handler","msg":"","name":"tsscplatform-jenkins","namespace":"devsecops","gvk":"redhatgov.io/v1alpha1, Kind=TsscPlatform","event_type":"runner_on_failed","job":"8313431429913368787","EventData.Task":"set_fact","EventData.TaskArgs":"","EventData.FailedTaskPath":"/opt/ansible/roles/tssc-platform/deployment/tasks/main.yml:99","error":"[playbook task failed]","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tpkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128\ngithub.com/operator-framework/operator-sdk/internal/ansible/events.loggingEventHandler.Handle\n\tsrc/github.com/operator-framework/operator-sdk/internal/ansible/events/log_events.go:87"}
{"level":"error","ts":1613766230.789223,"logger":"runner","msg":"\u001b[0;34mansible-playbook 2.9.11\u001b[0m\r\n\u001b[0;34m  config file = /etc/ansible/ansible.cfg\u001b[0m\r\n\u001b[0;34m  configured module search path = ['/usr/share/ansible/openshift']\u001b[0m\r\n\u001b[0;34m  ansible python module location = /usr/local/lib/python3.6/site-packages/ansible\u001b[0m\r\n\u001b[0;34m  executable location = /usr/local/bin/ansible-playbook\u001b[0m\r\n\u001b[0;34m  python version = 3.6.8 (default, Dec  5 2019, 15:45:45) [GCC 8.3.1 20191121 (Red Hat 8.3.1-5)]\u001b[0m\r\n\u001b[0;34mUsing /etc/ansible/ansible.cfg as config file\u001b[0m\r\n\r\nPLAYBOOK: tssc.yml *************************************************************\n\u001b[0;34m1 plays in /opt/ansible/playbooks/tssc.yml\u001b[0m\n\r\nPLAY [localhost] ***************************************************************\n\r\nTASK [Gathering Facts] *********************************************************\r\n\u001b[1;30mtask path: /opt/ansible/playbooks/tssc.yml:2\u001b[0m\n\u001b[0;32mok: [localhost]\u001b[0m\n\u001b[0;34mMETA: ran handlers\u001b[0m\n\r\nTASK [Get cluster dns] *********************************************************\r\n\u001b[1;30mtask path: /opt/ansible/playbooks/tssc.yml:4\u001b[0m\n\u001b[0;32mok: [localhost] => {\"changed\": false, \"resources\": [{\"apiVersion\": \"config.openshift.io/v1\", \"kind\": \"DNS\", \"metadata\": {\"creationTimestamp\": \"2021-01-18T07:21:45Z\", \"generation\": 1, \"managedFields\": [{\"apiVersion\": \"config.openshift.io/v1\", \"fieldsType\": \"FieldsV1\", \"fieldsV1\": {\"f:spec\": {\".\": {}, \"f:baseDomain\": {}}, \"f:status\": {}}, \"manager\": \"cluster-bootstrap\", \"operation\": \"Update\", \"time\": \"2021-01-18T07:21:45Z\"}], \"name\": \"cluster\", \"resourceVersion\": \"487\", \"selfLink\": \"/apis/config.openshift.io/v1/dnses/cluster\", \"uid\": \"46dd403e-00d7-46b6-84b1-6708ac00988a\"}, \"spec\": {\"baseDomain\": \"ocp4.example.com\"}, \"status\": {}}]}\u001b[0m\n\r\nTASK [set_fact] ****************************************************************\r\n\u001b[1;30mtask path: /opt/ansible/playbooks/tssc.yml:12\u001b[0m\n\u001b[0;32mok: [localhost] => {\"ansible_facts\": {\"cluster_name\": \"ocp4\"}, \"changed\": false}\u001b[0m\n\r\nTASK [set_fact] ****************************************************************\r\n\u001b[1;30mtask path: /opt/ansible/playbooks/tssc.yml:15\u001b[0m\n\u001b[0;32mok: [localhost] => {\"ansible_facts\": {\"openshift_base_domain\": \"example.com\", \"rhedsord_registry_password\": \"TODO\", \"rhedsord_registry_username\": \"TODO\", \"tssc_namespace\": \"devsecops\"}, \"changed\": false}\u001b[0m\n\r\nTASK [Set project names] *******************************************************\r\n\u001b[1;30mtask path: /opt/ansible/playbooks/tssc.yml:21\u001b[0m\n\u001b[0;32mok: [localhost] => {\"ansible_facts\": {\"argocd_project_name\": \"devsecops\", \"crw_deployment_type\": \"tssc\", \"crw_project_name\": \"devsecops\", \"gitea_project_name\": \"devsecops\", \"jenkins_project_name\": \"devsecops\", \"mattermost_namespace\": \"devsecops\", \"nexus_project_name\": \"devsecops\", \"quay_project_name\": \"devsecops\", \"rhsso_project_name\": \"devsecops\", \"selenium_grid_project_name\": \"devsecops\", \"sonarqube_project_name\": \"devsecops\"}, \"changed\": false}\u001b[0m\n\r\nTASK [Set tssc service account secret] *****************************************\r\n\u001b[1;30mtask path: /opt/ansible/playbooks/tssc.yml:35\u001b[0m\n\u001b[0;32mok: [localhost] => {\"ansible_facts\": {\"tssc_service_account_secret\": \"tssc-service-account-credentials\"}, \"changed\": false}\u001b[0m\n\r\nTASK [Look for default defined common variables] *******************************\r\n\u001b[1;30mtask path: /opt/ansible/playbooks/tssc.yml:39\u001b[0m\n\u001b[0;32mok: [localhost] => {\"ansible_facts\": {\"_tmp_parent\": \"{{ \\\"/\\\".join([ playbook_dir, \\\"../tmp\\\" ])|realpath }}\", \"common_included\": true, \"connected_cluster\": true, \"full_cluster_name\": \"{{ \\\".\\\".join([ cluster_name, openshift_base_domain ]) }}\", \"generated_password\": \"{{ lookup('password', '../tmp/rhssopasswd.txt length=15 chars=ascii_letters') }}\", \"manual_users\": [{\"admin\": true, \"clientRoles\": {\"account\": [\"manage-account\"], \"realm-management\": [\"manage-users\"]}, \"email\": \"realmadmin@example.com\", \"firstName\": \"OpenShift\", \"lastName\": \"Administrator\", \"password\": \"RedHatAdmin1\", \"realmRoles\": [\"offline_access\"], \"username\": \"openshift-admin\"}, {\"clientRoles\": {\"account\": [\"manage-account\"], \"realm-management\": [\"manage-users\"]}, \"email\": \"realmadmin@example.com\", \"firstName\": \"Realm\", \"lastName\": \"Administrator\", \"password\": \"{{ generated_password }}\", \"realmRoles\": [\"offline_access\"], \"username\": \"realmadmin\"}], \"number_of_users\": \"3\", \"oc_cli\": \"/usr/local/bin/oc\", \"openshift_users\": \"{{ manual_users + [workshop_admin] + sequence_users|from_yaml }}\", \"rhsso_app_label\": \"rhsso\", \"rhsso_project_name\": \"rhsso\", \"rhsso_realm_name\": \"openshift\", \"rhsso_users\": \"{{ openshift_users }}\", \"sequence_users\": \"{%- for username in lookup(\\\"sequence\\\", \\\"1-\\\" + number_of_users + \\\":user%0i\\\", wantlist=True) %}\\n  - username: {{ username }}\\n    firstName: {{ username }}\\n    lastName: Workshop user\\n    password: \\\"{{ generated_password }}\\\"\\n    email: {{ username }}@example.com\\n    jenkinsRole: edit\\n    nexusRole: nx-admin\\n{% endfor -%}\\n\", \"tmp_dir\": \"{{ \\\"/\\\".join([ _tmp_parent, full_cluster_name ]) }}\", \"trusted_router_certificates\": true, \"workshop_admin\": {\"admin\": true, \"email\": \"workshop@example.com\", \"firstName\": \"workshop-admin\", \"jenkinsRole\": \"admin\", \"lastName\": \"Workshop Admin\", \"password\": \"RedHatAdmin1\", \"username\": \"workshop-admin\"}, \"workshop_users\": \"{{ [workshop_admin] + sequence_users|from_yaml }}\"}, \"ansible_included_var_files\": [\"/opt/ansible/playbooks/../vars/tssc-platform/common.yml\"], \"changed\": false}\u001b[0m\n\r\nTASK [fail] ********************************************************************\r\n\u001b[1;30mtask path: /opt/ansible/playbooks/tssc.yml:43\u001b[0m\n\u001b[0;36mskipping: [localhost] => {\"changed\": false, \"skip_reason\": \"Conditional result was False\"}\u001b[0m\n\u001b[1;30mtask path: /opt/ansible/playbooks/tssc.yml:47\u001b[0m\r\nTASK [Look for default defined rhedsord variables] *****************************\r\n\u001b[1;30mtask path: /opt/ansible/playbooks/tssc.yml:47\u001b[0m\n\u001b[0;32mok: [localhost] => {\"ansible_facts\": {\"github_project\": \"jharmison-redhat\", \"rhedsord_included\": true, \"rhedsord_registry\": \"registry.sparta.spo.io:5000\"}, \"ansible_included_var_files\": [\"/opt/ansible/playbooks/../vars/tssc-platform/rhedsord.yml\"], \"changed\": false}\u001b[0m\n\r\nTASK [fail] ********************************************************************\r\n\u001b[1;30mtask path: /opt/ansible/playbooks/tssc.yml:51\u001b[0m\n\u001b[0;36mskipping: [localhost] => {\"changed\": false, \"skip_reason\": \"Conditional result was False\"}\u001b[0m\n\r\nTASK [include_tasks] ***********************************************************\r\n\u001b[1;30mtask path: /opt/ansible/playbooks/tssc.yml:55\u001b[0m\n\u001b[0;36mskipping: [localhost] => {\"changed\": false, \"skip_reason\": \"Conditional result was False\"}

Any idea?

**Comment by andykrohg** [comment]
Hey @piggyvenus, looks like your pull secret isn't setup correctly. The `TsscPlatform` expects a `Secret` with the name `pull-secret` in the namespace `openshift-config` by default, by you can optionally override it in the `tsscPlatform.pullSecret` attribute of your `TsscPlatform` CR. See Step 8 of [these instructions](https://github.com/ploigos/ploigos-software-factory-operator#tssc-demo-with-a-reference-app)

**Comment by piggyvenus** [comment]
I created the "pull-secret" secret in openshift-config in the form where I create the TSSCPlatform CR. 
let me try again. Thanks!
apiVersion: redhatgov.io/v1alpha1
kind: TsscPlatform
...
  creationTimestamp: '2021-02-19T16:58:36Z'
  generation: 1
  managedFields:
...
  namespace: devsecops
  finalizers:
    - finalizer.tsscplatform.redhatgov.io
spec:
  tsscPlatform:
    providers:
      ciTool: jenkins
    pullSecret:
      name: openshift-config
      namespace: pull-secret

**Comment by andykrohg** [comment]
Can you double check that? Looks like your `name` and `namespace` attributes are switched 🙃

**Comment by piggyvenus** [comment]
AH... right! thank you! My bad! Retest.
